Vector.cpp

```diff
@@ -173,66 +201,224 @@ class Custom_Vector
         }
         _size--;
     }
+
+    iterator begin() { return buffer; }
+    const_iterator begin() const { return buffer; }
+
+    iterator end() { return buffer + _size; }
+    const_iterator end() const { return buffer + _size; }
+
+    iterator cbegin() { return buffer + _size - 1; }
+    const_iterator cbegin() const { return buffer + _size - 1; }
+
+    iterator cend() { return buffer - 1; }
+    const_iterator cend() const { return buffer - 1; }
 };
 
-int main()
+TEST(Vector_Test, constructorByDefault)
 {
-    Custom_Vector a = {1, 4, 7, 9};
-    assert(a[0] == 1);
-    assert(a.at(2) == 7);
-    // assert(a.at(5));//выкинет исключение
-    assert(a.size() == 4);
-    assert(a.capacity() == 4);
-    a.push_back(5);
-    assert(a.size() == 5);
-    assert(a.capacity() == 8);
-    assert(a[4] == 5);
-    Custom_Vector b = a;
-    for (size_t i = 0; i < b.size(); i++)
-    {
-        std::cout << b[i] << " ";
-    }
-    std::cout << std::endl;
-    b.clear();
-    for (size_t i = 0; i < a.size(); i++)
-    {
-        std::cout << a[i] << " ";
-    }
-    std::cout << std::endl;
-    a.erase(2);
-    for (size_t i = 0; i < a.size(); i++)
-    {
-        std::cout << a[i] << " ";
-    }
-    std::cout << std::endl;
-    a.clear();
-    Custom_Vector c;
-    c.assign(4, 9);
-    for (size_t i = 0; i < c.size(); i++)
-    {
-        std::cout << c[i] << " ";
-    }
-    std::cout << std::endl;
-    c.assign(2, 7);
-    for (size_t i = 0; i < c.size(); i++)
-    {
-        std::cout << c[i] << " ";
-    }
-    std::cout << std::endl;
-    c.assign({1, 2, 3, 4, 5, 6});
-    for (size_t i = 0; i < c.size(); i++)
-    {
-        std::cout << c[i] << " ";
-    }
-    std::cout << std::endl;
-    c.erase(2);
-    for (size_t i = 0; i < c.size(); i++)
-    {
-        std::cout << c[i] << " ";
-    }
-    std::cout << std::endl;
-    c.clear();
-    assert(c.size() == 0);
-    assert(c.capacity() == 6);
-    return 0;
+    Custom_Vector test;
+
+    EXPECT_EQ(test.size(), 0);
+    EXPECT_EQ(test.capacity(), 0);
+}
+
+TEST(Vector_Test, constructorBySize)
```

Comment:
The following cases must be covered:
- input is empty
- input is not empty

Check the values here. You can write both in the same `TEST`.

Also, verify other tests cover empty case.